traffic_ops/traffic_ops_golang/cachesstats/cachesstats.go

```diff
@@ -146,11 +147,25 @@ func addStats(cacheData []CacheData, stats tc.Stats) []CacheData {
 		}
 		bandwidth, ok := stat.Stats[tc.StatNameBandwidth]
 		if ok && len(bandwidth) > 0 {
-			cache.KBPS = bandwidth[0].Val.(uint64)
+			if kbps, ok := bandwidth[0].Val.(string); !ok {
+				log.Warnf("couldn't convert %v into string", bandwidth[0].Val)
```

Comment:
These warnings would be hard to troubleshoot with this information. I think it would be helpful to include the `cache.HostName`, which stat we're actually trying to convert, and what URL was used to fetch the data from. For that last one, the URL is actually returned via `GetCacheStats` and `GetLegacyCacheStats`, but we ignore it. Maybe we should pass it into this function and use it in the warnings instead.

Reply:
Done